advent_of_code/2024/src/t/day06.clj

```diff
@@ -0,0 +1,65 @@
+(ns t.day06
+  (:require [clojure.core.match :refer [match]]
+            [clojure.string :as string]
+            [t.lib :as lib :refer [->long]]))
+
+(defn parse
+  [lines]
+  {:map (->> lines
+             (mapv (fn [line]
+                    (mapv (fn [c] (case c
+                                    \. true
+                                    \^ true
+                                    \# false))
+                          line))))
+   :guard [(->> (for [y (range (count lines))
+                      :let [line (nth lines y)]
+                      x (range (count line))
+                      :let [c (nth line x)]
+                      :when (= c \^)]
+                  [y x])
+                first)
+           [-1 0]]})
+
+(def turn
+  {[-1 0] [0 1]
+   [0 1] [1 0]
+   [1 0] [0 -1]
+   [0 -1] [-1 0]})
+
+(defn part1
+  [{m :map, [p d] :guard}]
+  (loop [[y0 x0] p
+         [dy dx] d
+         cells #{p}]
+    (let [[y x] [(+ y0 dy) (+ x0 dx)]]
+      (case (get-in m [y x] :out)
+        true (recur [y x] [dy dx] (conj cells [y x]))
+        false (recur [y0 x0] (turn [dy dx]) cells)
+        :out (count cells)))))
+
+(defn part2
+  [{m :map, [p d] :guard}]
+  (->> (for [obs-y (range (count m))
+               obs-x (range (count (m obs-y)))
+               :when (not= [obs-y obs-x] p)]
+         [obs-y obs-x])
+       (map (fn [p] (prn p) p))
```

Comment:
a little print left-over ?
```suggestion

```

Reply:
Yes; this one takes a long time to run (~25s) so I wanted to see progress. And then I disabled it so I never saw the print again.